src/components/feedback/Modal.tsx

```diff
@@ -0,0 +1,136 @@
+import classnames from 'classnames';
+import type { RefObject } from 'preact';
+import { useEffect, useLayoutEffect } from 'preact/hooks';
+
+import { useElementShouldClose } from '../../hooks/use-element-should-close';
+import { useSyncedRef } from '../../hooks/use-synced-ref';
+import { useUniqueId } from '../../hooks/use-unique-id';
+
+import type { PresentationalProps } from '../../types';
+import { downcastRef } from '../../util/typing';
+
+import Overlay from '../layout/Overlay';
+import Panel from '../layout/Panel';
+import type { PanelProps } from '../layout/Panel';
+
+type ComponentProps = {
+  /**
+   * Modal _should_ be provided with a close handler. We have a few edge use
+   * cases, however, in which we need to render a "non-closeable" modal.
+   */
+  onClose?: () => void;
+  width?: 'sm' | 'md' | 'lg' | 'custom';
+  /**
+   * Element that should take focus when the Dialog is first rendered. When not
+   * provided, the dialog's outer element will take focus. Setting this prop to
+   * `null` will opt out of focus routing.
+   */
+  initialFocus?: RefObject<HTMLElement | null> | null;
```

Comment:
I see this prop handles three possible "states":

* A ref is provided, in which case the referenced component gets focused.
* The prop is not provided at all, in which case the modal itself gets focused.
* The `null` value is provided, in which case we opt out from focusing capabilities.

I think the last two can be easily confused in consumer code. If I see `<Modal initialFocus={null} ... />` I will not remember that means the focusing is disabled.

I don't see an easy solution though. Maybe accepting a constant value instead, which is more clear in consumer code? Something in the lines of `<Modal initialFocus="DISABLE_FOCUS" ... />`. It's a bit hacky, but maybe more descriptive.

Other ideas?

Reply:
> The prop is not provided at all, in which case the modal itself gets focused.
> The null value is provided, in which case we opt out from focusing capabilities.

What uses do we have for these two separate cases? Could we just always use one of the two behaviors here?

Reply:
Good feedback, thanks!